autoemulate/experimental/model_selection.py

```diff
@@ -97,6 +97,8 @@ def cross_validate(
 
         # fit model
         x, y = next(iter(train_loader))
+        # TODO: should we set random_state in the models?
+        # m = model(x, y, random_state=123, **best_model_config)
```

Comment:
1. Should we set `random_state` for all the models?
2. If so, may make sense to set a default value in `Emulator` base class rather than here

Reply:
The deeper question is, do we enforce determinism for AutoEmulate models, if not, should we? Looks like AutoEmulate v0 does set up cross validation with a random_state, but we currently don't do this in experimental/v1